core/localizer.js

```diff
@@ -497,120 +574,124 @@ var Localizer = exports.Localizer = Montage.specialize( /** @lends Localizer.pro
+            if (!_component) {
+                _component = this.component || rootComponent;
+
+                if (typeof _component.LocalizerDidChangeLocale === "function") {
+                    _component.LocalizerDidChangeLocale(this, previousLocale, this._locale);
                 }
-                defaultLocale = defaultLocale || window.navigator.userLanguage || window.navigator.language;
             }
 
-            defaultLocale = defaultLocale || "en";
-            this.locale = defaultLocale;
+            // Get the private `_childComponents` in order to avoid to create an empty array,
+            // indeed childComponents is set lazily.
+            var childComponents = _component._childComponents;
 
-            this.loadMessages();
+            if (childComponents) {
+                var child;
 
-            return this;
-        }
-    },
+                for (var i = 0; i < childComponents.length; i++) {
+                    child = childComponents[i];
 
-    _delegate: {
-        value: null
-    },
+                    if (typeof child.LocalizerDidChangeLocale === "function") {
+                        child.LocalizerDidChangeLocale(this, previousLocale, this._locale);
+                    }
 
-    /**
-     * Delegate to get the default locale.
-     * Should implement a `getDefaultLocale` method that returns
-     * a language-tag string that can be passed to {@link locale}
-     * @type Object
-     * @default null
-     */
-    delegate: {
-        get: function () {
-            return this._delegate;
-        },
-        set: function (value) {
-            if (this._delegate !== value) {
-                this._delegate = value;
-                this.init();
+                    if (child._childComponents) {
+                        this._dispatchLocaleChange(previousLocale, child);
+                    }
+                }
             }
         }
-    },
+    }
 
-    locale: {
-        get: function () {
-            return this._locale;
-        },
-        set: function (value) {
-            try {
-                Object.getPropertyDescriptor(Localizer.prototype, "locale").set.call(this, value);
-            } catch (e) {
-                value = "en";
-                Object.getPropertyDescriptor(Localizer.prototype, "locale").set.call(this, value);
-            }
+}, {
 
-            // If possible, save locale
-            if (typeof window !== "undefined" && window.localStorage) {
-                try {
-                    window.localStorage.setItem(LOCALE_STORAGE_KEY, value);
-                } catch (e) {
-                    // LocalStorage quota might have been exceeded
-                    // iOS Safari emits a quota exceeded error from private mode always
-                }
+    defaultLocalizer: {
+        value: function () {
+            if (!defaultLocalizer) {
+                defaultLocalizer = new Localizer();
+                defaultLocalizer.shouldStoreLocale = true;
```

Comment:
shouldStoreLocale should be storesLocale, shouldStoreLocale would be a proper name for a delegate method